pkg/config/config_parser.go

```diff
@@ -102,10 +102,43 @@ func ValidateCQBlock(cq *CloudQuery) diag.Diagnostics {
 	return diags
 }
 
+func ProcessValidateProviderBlock(plist []*Provider) (Providers, diag.Diagnostics) {
+	var diags diag.Diagnostics
+	existingProviders := make(map[string]struct{}, len(plist))
+
+	var ret Providers
+
+	for _, v := range plist {
+		if v.Alias != "" {
+			if _, ok := existingProviders[v.Alias]; ok {
+				diags = diags.Add(diag.FromError(fmt.Errorf("provider with alias %s for provider %s already exists, give it a different alias", v.Alias, v.Name), diag.USER, diag.WithSummary("Duplicate Alias")))
+				continue
+			}
+			existingProviders[v.Alias] = struct{}{}
+		} else {
+			if _, ok := existingProviders[v.Name]; ok {
+				diags = diags.Add(diag.FromError(fmt.Errorf("provider with name %s already exists, use alias in provider configuration block", v.Name), diag.USER, diag.WithSummary("Provider Alias Required")))
+				continue
+			}
+			existingProviders[v.Name] = struct{}{}
+			v.Alias = v.Name
+		}
+		var err error
+		v.Configuration, err = yaml.Marshal(v.ConfigKeys)
+		if err != nil {
+			diags = diags.Add(diag.FromError(err, diag.INTERNAL, diag.WithSummary("ConfigKeys marshal failed")))
```

Comment:
[non-blocking] Should this be `diag.User` too?

Reply:
I don't think so, it would mean yaml is acting weirdly or we changed ConfigKeys type to be something else etc? or there's something inside the interface{} that we don't handle but we'd ideally like to?

Reply:
Ah cool, thanks for explaining. Lets keep it this way and see what gets reported